### PR TITLE
Userstore

### PR DESCRIPTION
**Pull request by Anurodhyadav**
Please review it..

### CODE REVIEW

**Comment by shakyaprasen** [comment]
> Please review it..

Please write relevant and useful information in the description box.
What changes did you do?
What did these changes address?
Any issues while doing these changes?

File: src/components/navBar.vue [hunk 1]
```diff
@@ -1,61 +1,156 @@
...
+      data-toggle="collapse"
+      data-target="#navbarSupportedContent"
+      aria-controls="navbarSupportedContent"
+      aria-expanded="false"
+      aria-label="Toggle navigation"
+    >
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto"></ul>
```
Comment (shakyaprasen) [inline src/components/navBar.vue:18]: removed unused code

File: src/components/navBar.vue [hunk 2]
```diff
@@ -1,61 +1,156 @@
...
+          <a
+            class="nav-link dropdown-toggle"
+            href="#"
+            id="navbarDropdown"
+            role="button"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+          >
+            <span class="navEmail" v-if=" user_photoUrl == null">{{ current_user_email }}</span>
```
Comment (shakyaprasen) [inline src/components/navBar.vue:38]: please don't use weak equals operator `==` use `===` or just `v-if=user_photoUrl`

File: src/components/navBar.vue [hunk 3]
```diff
@@ -1,61 +1,156 @@
...
+            <img v-else class="navimg"
+              v-bind:src="user_photoUrl"
+              width="30"
+              height="30"
+            />
+          </a>
+          <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+            <div class="card text-white">
+              <span class="navEmail" v-if=" user_photoUrl == null"></span>
```
Comment (shakyaprasen) [inline src/components/navBar.vue:47]: same as above

File: src/components/navBar.vue (hunk 2 above)
Comment (shakyaprasen) [reply src/components/navBar.vue:38]: no camelcasing for variable names. 
user_photoUrl 
current_user_email

File: src/components/navBar.vue [hunk 4]
```diff
@@ -1,61 +1,156 @@
...
+            href="#"
+            id="navbarDropdown"
+            role="button"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+          >
+            <span class="navEmail" v-if=" user_photoUrl == null">{{ current_user_email }}</span>
+            <img v-else class="navimg"
+              v-bind:src="user_photoUrl"
```
Comment (shakyaprasen) [inline src/components/navBar.vue:40]: no camelcasing for variable names.

File: src/components/navBar.vue [hunk 5]
```diff
@@ -1,61 +1,156 @@
...
+              width="30"
+              height="30"
+            />
+          </a>
+          <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+            <div class="card text-white">
+              <span class="navEmail" v-if=" user_photoUrl == null"></span>
+              <img v-else class="dropdnimg" v-bind:src="user_photoUrl"
```
Comment (shakyaprasen) [inline src/components/navBar.vue:48]: use camelcasing for variable names.

File: src/components/navBar.vue [hunk 6]
```diff
@@ -1,61 +1,156 @@
...
+                width="90" height="90"
+              />
+                <h6 class="card-title">{{ user_name }}</h6>
```
Comment (shakyaprasen) [inline src/components/navBar.vue:51]: use camelcasing for variable names.

File: src/components/navBar.vue [hunk 7]
```diff
@@ -1,61 +1,156 @@
...
+                <p>{{ current_user_email }}</p>
```
Comment (shakyaprasen) [inline src/components/navBar.vue:52]: use camelcasing for variable names.

File: src/components/navBar.vue [hunk 8]
```diff
@@ -1,61 +1,156 @@
...
+      user_name: '',
+      user_photoUrl: '',
     };
   },
   methods: {
     async signOut() {
       try {
-        firebase.auth().signOut();
+        const data = firebase.auth().signOut();
+        console.log(data);
```
Comment (shakyaprasen) [inline src/components/navBar.vue:81]: console.log

File: src/router/index.js [hunk 9]
```diff
@@ -14,7 +14,7 @@ import SignIn from '../views/SignIn.vue';
 import Review from '../views/Review.vue';
 import Test from '../views/Test.vue';
 import Result from '../views/Result.vue';
-import Admin from '../views/TempAdmin.vue';
+import Adminregister from '../views/Adminregister.vue';
```
Comment (shakyaprasen) [inline src/router/index.js:17]: AdminRegister is proper CapitalCasing

File: src/router/index.js [hunk 10]
```diff
@@ -113,6 +113,7 @@ const router = new VueRouter({
 router.beforeEach((to, from, next) => {
   const requiresAuth = to.matched.some((record) => record.meta.requiresAuth);
   const isAuthenticated = firebase.auth().currentUser;
+  // const isSuperUser = (isAuthenticated.email === 'nagma@gmail.com');
```
Comment (shakyaprasen) [inline src/router/index.js:116]: unused code

File: src/router/index.js [hunk 11]
```diff
@@ -113,6 +113,7 @@ const router = new VueRouter({
...
   if (requiresAuth && !isAuthenticated) {
     // alert('Please signin first');
```
Comment (shakyaprasen) [inline src/router/index.js:118]: remove unused code

File: src/views/Adminregister.vue [hunk 12]
```diff
@@ -0,0 +1,227 @@
+<template>
+<div v-if="allowAccess">
+<NavBar/>
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:3]: missing proper indentation

File: src/views/Adminregister.vue [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+  <div class="signup-form">
+      <h2>AdminRegister</h2>
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:5]: use proper indentation

File: src/views/Adminregister.vue [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+            name="email"
+            v-model="email"
+            placeholder="Email Address"
+            required="required"
+          />
+        </div>
+      </div>
+  </div>
+      <div class="form-group">
+        <button @click="registerNew()" type="submit" class="btn btn-primary btn-lg">
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:27]: no need to call the function just write `registerNew`

File: src/views/Adminregister.vue [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+            AddAdmin
+        </button>
+      </div>
+  <table class="table table-borderless mx-auto m-2 text-left"
+              style="width:40%; color:aliceblue; font-weight:bold">
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:32]: class preferable instead of style

File: src/views/Adminregister.vue [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+      emailList: [],
+      myEmail: '',
+      myEmailId: '',
+      allowAccess: false,
+      mySuperuser: '',
+    };
+  },
+  methods: {
+    async removeAdmin(emailItem) {
+      await db.collection('UserData').where('email', '==', emailItem).get().then((mySnapshot) => {
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:70]: use strict equality checker `===`

File: src/views/Adminregister.vue [hunk 17]
```diff
@@ -0,0 +1,227 @@
...
+          };
+          this.myEmailId = myData.id;
+          db.collection('UserData').doc(this.myEmailId).set({
+            role: '',
+          },
+          { merge: true });
+        });
+      });
+      this.acceptAlert('danger', 'Permission chnaged', 'Admin removed');
+      await window.location.reload();
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:83]: why await reload?

File: src/views/Adminregister.vue [hunk 18]
```diff
@@ -0,0 +1,227 @@
...
+        if (this.myUserId) {
+          this.email = '';
+          this.acceptAlert('success', 'Admin Regitered', 'The admin was successfully completed');
+        } else {
+          this.acceptAlert('danger', 'User doesnot exits', 'The user with given email address doesnot exists');
+        }
+      } catch (e) {
+        this.acceptAlert('danger', 'Error', e);
+      }
+      await window.location.reload();
```
Comment (shakyaprasen) [inline src/views/Adminregister.vue:108]: no need to await reload

File: src/views/Adminregister.vue (hunk 17 above)
Comment (shakyaprasen) [reply src/views/Adminregister.vue:83]: can it be done using vue router itself?

File: src/views/Files.vue [hunk 19]
```diff
@@ -93,6 +96,23 @@ export default {
...
   },
+  created() {
+    const user = firebase.auth().currentUser;
+    db.collection('UserData').where('email', '==', user.email).get().then((snapshot) => {
+      snapshot.forEach((doc) => {
+        const Data = {
+          role: doc.data().role,
+        };
+        this.userRole = Data.role;
+        console.log(this.userRole);
```
Comment (shakyaprasen) [inline src/views/Files.vue:107]: console log

File: src/views/Test.vue [hunk 20]
```diff
@@ -46,8 +46,12 @@ export default {
   },
   data() {
     return {
-      instructionsPop: true,
+      popupActivo: true,
+      current_User_email: '',
```
Comment (shakyaprasen) [inline src/views/Test.vue:50]: please use consistent casing, there is camel case someplace and snake case in others

File: src/views/Test.vue [hunk 21]
```diff
@@ -57,16 +61,42 @@ export default {
...
+              transcription: this.newItem,
+              user: this.current_User_email,
             });
             this.acceptAlert('success', 'File Saved', 'The test was successfully completed.');
             this.$vs.loading.close();
             this.$router.push({ name: 'Result' });
           } catch (e) {
             alert('Error');
           }
+          db.collection('transData').where('user', '==', this.current_User_email).get().then((snapshot) => {
```
Comment (shakyaprasen) [inline src/views/Test.vue:74]: strict equality operator `===`

File: src/views/Test.vue [hunk 22]
```diff
@@ -57,16 +61,42 @@ export default {
...
+              };
+              this.myId = Data.id;
+              this.transId.push(this.myId);
+              db.collection('UserData').where('Email', '==', this.current_User_email).get().then((mySnapshot) => {
+                mySnapshot.forEach((myDoc) => {
+                  const myData = {
+                    id: myDoc.id,
+                  };
+                  this.myUserId = myData.id;
+                  console.log(this.myUserId);
```
Comment (shakyaprasen) [inline src/views/Test.vue:87]: console.log here

File: src/views/Upload.vue [hunk 23]
```diff
@@ -16,7 +16,8 @@
             placeholder="Transcription" v-model="newComment"></textarea>
         </div>
     </form>
-    <b-button type="button" class="btn btn-info m-1" @click="addItem()">Save</b-button>
+   <b-button type="button" class="btn btn-info m-1"
+     @click="addItem()">Save</b-button>
```
Comment (shakyaprasen) [inline src/views/Upload.vue:20]: `addItem` would also work no need to call the function at the start

File: src/views/Upload.vue [hunk 24]
```diff
@@ -71,6 +76,23 @@ export default {
       this.audioName = audioName;
     },
   },
+  created() {
+    const user = firebase.auth().currentUser;
+    db.collection('UserData').where('email', '==', user.email).get().then((snapshot) => {
```
Comment (shakyaprasen) [inline src/views/Upload.vue:81]: strict equality operator here

File: src/views/Upload.vue [hunk 25]
```diff
@@ -71,6 +76,23 @@ export default {
...
+      snapshot.forEach((doc) => {
+        const Data = {
+          role: doc.data().role,
+        };
+        this.userRole = Data.role;
+        console.log(this.userRole);
```
Comment (shakyaprasen) [inline src/views/Upload.vue:87]: console log here

**Review by shakyaprasen — CHANGES_REQUESTED** [review]
Before copy pasting code, please check if there are bits of code not needed.
A lot of console log is still left due to this.

File: src/views/Test.vue (hunk 21 above)
Comment (Anurodhyadav) [reply src/views/Test.vue:74]: using strict equality operator doesnot work

File: src/views/Adminregister.vue (hunk 17 above)
Comment (Anurodhyadav) [reply src/views/Adminregister.vue:83]: await has been removed to setTimeout for 1.2 seconds as created need to be loaded every time to represent updated data.

**Pull request merged**